File: __prologue_server/ecs/comps/point.go
/* */

package comps

type Point struct {
	X int
	Y int
}

func NewPoint(x, y int) Point {
	return Point{
		X: x,
		Y: y,
	}
}

func (p Point) GetCoords() (x, y int) {
	return p.X, p.Y
}

File: __prologue_client/core/mapa/point.js
/* */

class Point {
  constructor(x, y) {
    this.x = x;
    this.y = y;
  }

  static getCoords(p) {
    return {
      x: p.x,
      y: p.y,
    };
  }
}

export {
  Point
};
